### DISCUSSION TITLE
Can sea orm generate database migrations from entity?

### DISCUSSION THREAD

**Discussion by xiayulu** [SeaQL/sea-orm#314] [Q&A] [answered]
Hello, I am using sea orm building an app.

I find it hard to write migrations with SQL by myself.

Can sea orm generate database migrations from entity? That will simplify rust web development a lot.

**Comment by tyt2y3** [comment, marked as answer]
In short, yes. 

Take a look at https://github.com/SeaQL/sea-orm-rocket-starter/blob/master/src/db/migrations/mod.rs

Ref: https://docs.rs/sea-orm/0.x/sea_orm/schema/struct.Schema.html#method.create_table_from_entity

**Reply by Eugeny** [reply]
This doesn't work if the entity struct is ever changed after the fact, right?

**Reply by billy1624** [reply]
Yes, you need migration if your DB schema keeps changing and evolving

**Reply by gak** [reply]
Is there an issue for this? Mature ORMs in other languages do this, e.g. Rails and Django. They do a diff between the current schema migration vs the new models, then generate a new migration.

**Reply by billy1624** [reply]
Hey @gak, at this stage, we have limited developing capacity thus would want to focus on stabilizing the API and offer first tier support to SQLite, MySQL and Postgres.

**Reply by Baiyuetribe** [reply]
@tyt2y3 The example above is only suitable for cases where the table structure remains unchanged. Once the table structure changes, it will become invalid.
```rust
schema.create_table_from_entity(entity).if_not_exists() // invalid if entity changed
```
Wish to add this new method `alter_table_from_entity `
```rust
schema.alter_table_from_entity(entity).if_not_exists()
```
